lib/spack/spack/detection/common.py

```diff
@@ -235,9 +269,44 @@ def compute_windows_program_path_for_package(pkg):
         return []
     # note windows paths are fine here as this method should only ever be invoked
     # to interact with Windows
-    program_files = "C:\\Program Files{}\\{}"
+    program_files = "{}:\\Program Files{}\\{}"
+    drive_letter = os.environ['WINDIR'][0]
```

Comment:
Yes, these were different iterations of development. I hadn't realized I was using a different env var good catch, I'll consolidate that. I'm also thinking python must have a way to detect current drive letter so perhaps this can be made more robust as well.